File: hd_wallet_js/router.js
import { generate, restore } from "./routeHandlers.js";

export default (event) => {
  if (
    event.requestContext.http.path == "/wallet/generate" &&
    event.requestContext.http.method == "GET"
  ) {
    return generate(event.queryStringParameters);
  } else if (
    event.requestContext.http.path == "/wallet/restore" &&
    event.requestContext.http.method == "POST"
  ) {
    return restore(JSON.parse(event.body));
  } else {
    throw new Error(`${event.requestContext.http.path} match not found`);
  }
};

File: hd_wallet_go/wallet/router/router.go
package router

import (
	"fmt"
	"hd_wallet_go/wallet/handlers"

	"github.com/aws/aws-lambda-go/events"
)

type Router struct{}

func (r *Router) Route(req events.APIGatewayProxyRequest) (string, error) {
	if req.Path == "/wallet/generate" && req.RequestContext.HTTPMethod == "GET" {
		return handlers.Generate()
	} else if req.Path == "/wallet/restore" && req.RequestContext.HTTPMethod == "POST" {
		return handlers.Restore()
	} else {
		return "", fmt.Errorf("%s match not found", req.Path)
	}
}